src/main/java/com/iexec/common/utils/MultiAddressHelper.java

```diff
@@ -0,0 +1,24 @@
+package com.iexec.common.utils;
+
+import io.ipfs.multiaddr.MultiAddress;
+import org.web3j.utils.Numeric;
+
+public class MultiAddressHelper {
+
+    private static final String IPFS_GATEWAY_URI = "https://gateway.ipfs.io";
+
+    private MultiAddressHelper() {
+        throw new UnsupportedOperationException();
+    }
+
+    public static String convertToURI(String hexaString) {
+        // Currently, only the IPFS format is supported in the multiaddress field in the smart contract, so
+        // if the MultiAddress object can't be constructed, it means it is an HTTP address.
+        try {
+            MultiAddress address = new MultiAddress(Numeric.hexStringToByteArray(hexaString));
+            return IPFS_GATEWAY_URI + address.toString();
```

Comment:
Does it need a "slash" here in the middle?

Reply:
Just moved some code from Worker to Common, nothing changed (https://github.com/iExecBlockchainComputing/iexec-worker/blob/c5cf832c55f8eceb1daa90fe46f020741d315448/src/main/java/com/iexec/worker/utils/MultiAddressHelper.java#L20)

Reply:
ah okey!